sql-plugin/src/main/331+/scala/com/nvidia/spark/rapids/GpuCheckOverflowInTableInsert.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
+import com.nvidia.spark.rapids.shims.ShimUnaryExpression
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.rapids.shims.spark331plus.CastOverflowUtil
+import org.apache.spark.sql.types.DataType
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+/**
+ * Handle the new Expression added in Spark-3.3.1+.
+ * Casting a numeric value as another numeric type in store assignment. It captures the arithmetic
+ * exception thrown by Cast and shows a relevant error message.
+ * The current implementation in Spark creates a child of type Cast with AnsiEnabled set to true.
+ *
+ * The GPU equivalent columnar evaluation is a delegation to the child's columnar evaluation which
+ * is a GpuCast. In order to match Spark's Exception, we need to catch the Exception passing it to
+ * QueryExecutionError.
+ * The calculation of the sideEffect is delegated to the child Expression "GpuCast".
+ */
+case class GpuCheckOverflowInTableInsert(child: Expression, columnName: String)
+  extends ShimUnaryExpression with GpuExpression {
+
+  override def dataType: DataType = child.dataType
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    try {
+      child.columnarEval(batch)
+    } catch {
+      // Convert ArithmeticException to Spark error class "CAST_OVERFLOW_IN_TABLE_INSERT"
+      case _: ArithmeticException =>
```

Comment:
Why is the ArithmeticException and not SparkArithmeticException?

https://github.com/apache/spark/blob/7ae2b85c485a2f7ceb32543c64f08f3b1f120464/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L2371

Reply:
Changed to `SparkArithmeticException`.
Initially I opted to `ArithmeticException` because this is the way it is used across the plugin code and the flexibility of using the code outside spark package.